File: examples/php/04_bloom_filter.php
<?php
require 'vendor/autoload.php';

$client = new Predis\Client([
    'host' => '127.0.0.1',
    'port' => 6379,
]);

echo "--- 04 Bloom Filter ---\n";
$key = "bad_ips";

// BF.ADD
$client->executeRaw(['BF.ADD', $key, '192.168.1.55']);
$client->executeRaw(['BF.ADD', $key, '10.0.0.1']);

// BF.EXISTS
$check = '192.168.1.55';
$exists = $client->executeRaw(['BF.EXISTS', $key, $check]);
echo "Is '$check' bad? " . ($exists ? "Yes" : "No") . "\n";

$safe = '8.8.8.8';
$existsSafe = $client->executeRaw(['BF.EXISTS', $key, $safe]);
echo "Is '$safe' bad? " . ($existsSafe ? "Yes" : "No") . "\n";
?>

File: examples/go/04_bloom_filter.go
package main

import (
	"context"
	"fmt"

	"github.com/redis/go-redis/v9"
)

var ctx = context.Background()

func main() {
	rdb := redis.NewClient(&redis.Options{
		Addr: "localhost:6379",
	})

	fmt.Println("--- 04 Bloom Filter ---")
	key := "spam_emails"

	// BF.ADD
	rdb.Do(ctx, "BF.ADD", key, "spam@evil.com")
	rdb.Do(ctx, "BF.ADD", key, "virus@bad.com")

	// BF.EXISTS
	check := "spam@evil.com"
	exists, _ := rdb.Do(ctx, "BF.EXISTS", key, check).Int()

	fmt.Printf("Is '%s' blocked? %v\n", check, exists == 1)

	checkSafe := "friend@good.com"
	existsSafe, _ := rdb.Do(ctx, "BF.EXISTS", key, checkSafe).Int()
	fmt.Printf("Is '%s' blocked? %v\n", checkSafe, existsSafe == 1)
}
